### ISSUE TITLE
Unable to install the project as an external contributor

### ISSUE THREAD

**Issue by honzajavorek** [apify/apify-docs#909]
The repository uses private npm packages:

```
npm ERR! code E401
npm ERR! 401 Unauthorized - GET https://npm.pkg.github.com/download/@apify-packages/ui-components/0.86.0/6927975c9bcef7f899f41292e4b3d2b675011690 - authentication token not provided
```

I'm filing this as an issue, because I think it would be vital of anyone could clone this repo and contribute to it. Even if only some components of the docs could run locally, in a similar sense like this:

https://github.com/apify/apify-docs/blob/82d6da569455bdc5a8e8dece260431959e04f5ef/CONTRIBUTING.md?plain=1#L83

Currently, it is not possible to even successfully finish `npm install` of the repo.

**Comment by B4nan** [comment]
cc @jbartadev I lost track of this, do you still plan to make the package public or inline the necessary components here?

**Comment by B4nan** [comment]
@honzajavorek here are docs on how to install the package, but can't add you to the project myself

https://www.notion.so/apify/How-to-use-apify-packages-npm-registry-dfc20d3691ff4a879ef30ad2d104db75

**Comment by honzajavorek** [comment]
Thanks, but that just points to 404 😅

**Comment by B4nan** [comment]
Yeah, it's a completely valid point, we've been struggling with this internally too. This started relatively recently when a new homepage was added which uses the shared (and private) packages. I thought the packages were already made public, but as said above, I kinda lock track of the current state and plans.

**Comment by jbartadev** [comment]
The ui-components library is published as a public package, it should be possible to install it without any authorization

https://github.com/apify-packages/shared-components/pkgs/npm/ui-components

**Comment by B4nan** [comment]
Maybe we just need to update its version, but I remember there were breaking changes. Let me have a look.

**Comment by B4nan** [comment]
Build is passing with latest version, will update in a minute.

**Comment by B4nan** [comment]
Btw @jbartadev you might want to ship v1 of those public packages, as right now, every `0.x` release is considered major/breaking, so renovate won't update it automatically for us. If you would ship v1 and start distinguishing feature releases from major/breaking ones, it would help downstream to stay up to date.

**Comment by honzajavorek** [comment]
> The ui-components library is published as a public package, it should be possible to install it without any authorization
> 
> [https://github.com/apify-packages/shared-components/pkgs/npm/ui-components](https://github.com/apify-packages/shared-components/pkgs/npm/ui-components)
> 
> It's probably because there is outdated version of this package

When I click on [the link](https://github.com/apify-packages/shared-components/pkgs/npm/ui-components), I get 404.

**Comment by B4nan** [comment]
I don't think that's relevant, it is a private registry, it won't be public. The package is, you should be able to install it.

**Comment by B4nan** [comment]
Hmm, I think you still need to set up the registry (which is described in notion, to which you clearly don't have access 🙃).

<img width="494" alt="image" src="https://github.com/apify/apify-docs/assets/615580/c708c26d-abb4-46a3-84af-f7f08e26bb23">

But I guess you first need to be added to https://github.com/apify-packages which I don't have permission for.

@jbartadev how hard would it be to publish those to NPM so we can skip this completely?

**Comment by honzajavorek** [comment]
The `.npmrc` is commited to the repo, so that shouldn't be the problem. Without it npm wouldn't even know what to do with the `@apify-packages` prefix.

If I need to be added somewhere for the install to work, it means the package isn't effectively public I guess. In a sense that an external contributor won't be able to run the site locally.

**Comment by mnmkng** [comment]
Just ran into this even though I'm an owner in the `@apify-packages` org 😅 Plus I had to update the installation guide in Notion, because it was outdated.

Can we pls make sure that we only use publicly available deps in public repos? It's a shame that this is so hard to install. Let's publish the components to NPM or something. Thanks!

**Comment by honzajavorek** [comment]
Just a reminder I'm about to work on the repo this week and this is kind of a blocker to my contributions (I can contribute 'blindly' of course, but it's not ideal).

**Comment by mnmkng** [comment]
@honzajavorek I could add you to the Apify Packages GitHub org. But then this would never get fixed 😅

**Comment by honzajavorek** [comment]
I understand 😀 and I definitely do prefer a solution for all hypothetical contributors, not a solution for just me. Just sneakily using my special situation as a leverage 😅

**Comment by jbartadev** [comment]
Hey guys, we're trying to solve this issue. The package is public and the issue is [a feature](https://github.com/orgs/community/discussions/33875). We can't easily do anything with it with our current logic of package publication.

A simple solution should be to use your access token for installation.

**Comment by B4nan** [comment]
Could we just inline the necessary components directly to this repository for now, and deal with the publication later? Sounds like an afternoon job but maybe I am missing something.

**Comment by mnmkng** [comment]
@honzajavorek can you pls try if it works even when you don't have access to the org?

1. Create a token:

![](https://file.notion.so/f/f/0da99642-1074-4390-8fd7-7d735176986a/811654e6-f3a6-4bb3-b0fa-e3d9848e5bbf/Snimek_obrazovky_2020-11-12_v_15.16.45.png?id=4f02e333-c4f2-433d-a5c2-be9fe9bfaaec&table=block&spaceId=0da99642-1074-4390-8fd7-7d735176986a&expirationTimestamp=1713916800000&signature=RpwNRGrPrmaraWjXoy5rB-8bULmWlsZqWsooXBqS-6g&downloadName=Snimek_obrazovky_2020-11-12_v_15.16.45.png)

2. Log in to NPM:

```bash
npm login --scope=@apify-packages --registry=https://npm.pkg.github.com --auth-type=legacy
```

Maybe it will work even if you're not a member of the org.

**Comment by honzajavorek** [comment]
@jbartadev Thanks for investigating the issue! Sad to hear that this is how it works.

@mnmkng Will try and report back.

**Comment by honzajavorek** [comment]
The personal access token works! I got past `npm install` with success.

There are two types of personal access tokens now, new fine-grained personal access token, and _classic_ token. The screenshot you sent is from a setup of the _classic_ token. I tried to figure out how to do the same with the new kind of token, but I couldn't find anything related to packages on the setup page, so I quickly gave up and created a _classic_ token. Doesn't feel right to jump over ten warnings that it's not recommended anymore, but it works.
